flake.nix

```diff
@@ -2,64 +2,51 @@
   description = "A collection of Terraform versions that are automatically updated";
 
   inputs = {
-    flake-utils.inputs.systems.follows = "systems";
-    flake-utils.url = "github:numtide/flake-utils";
-    nixpkgs-unstable.url = "github:nixos/nixpkgs/nixpkgs-unstable";
     nixpkgs.url = "github:nixos/nixpkgs/nixos-23.05";
+    nixpkgs-unstable.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+
     systems.url = "github:nix-systems/default";
+    flake-parts.url = "github:hercules-ci/flake-parts";
   };
 
-  outputs = { self, flake-utils, nixpkgs-unstable, nixpkgs, ... }:
-    flake-utils.lib.eachDefaultSystem
-      (system:
-        let
-          pkgs = nixpkgs.legacyPackages.${system};
-          pkgs-unstable = nixpkgs-unstable.legacyPackages.${system};
-          versions = builtins.fromJSON (builtins.readFile ./versions.json);
-        in
-        {
-          devShells.default = pkgs.mkShell {
-            buildInputs = [
-              pkgs-unstable.black
-              (pkgs-unstable.python3.withPackages
-                (ps: [
-                  ps.pygithub
-                  ps.semver
-                ])
-              )
-              pkgs-unstable.nix-prefetch
-              pkgs.nodejs
-              pkgs.rubyPackages.dotenv
-            ];
-          };
-          # https://github.com/NixOS/nix/issues/7165
-          checks = self.packages.${system};
-          packages = builtins.mapAttrs
-            (version: { hash, vendorHash }: self.lib.buildTerraform {
-              inherit system version hash vendorHash;
-            })
-            versions;
-        }) // {
-      lib.buildTerraform = { system, version, hash, vendorHash }:
-        let
-          pkgs = nixpkgs.legacyPackages.${system};
-          pkgs-unstable = nixpkgs-unstable.legacyPackages.${system};
-        in
-        # https://www.hashicorp.com/blog/hashicorp-adopts-business-source-license
-        if builtins.compareVersions version "1.6.0" >= 0
-        then
-        # https://github.com/NixOS/nixpkgs/blob/nixpkgs-unstable/pkgs/applications/networking/cluster/terraform/default.nix
-          pkgs-unstable.mkTerraform
-            {
-              inherit version hash vendorHash;
-              patches = [ "${nixpkgs-unstable}/pkgs/applications/networking/cluster/terraform/provider-path-0_15.patch" ];
-            }
-        else
-        # https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/applications/networking/cluster/terraform/default.nix
-          pkgs.mkTerraform {
-            inherit version hash vendorHash;
-            patches = [ "${nixpkgs}/pkgs/applications/networking/cluster/terraform/provider-path-0_15.patch" ];
```

Comment:
I think this repository should provide its own patches. If tomorrow this patch gets removed from `nixpkgs`, this repository will fail to build after updating flake.lock.
So I added this patch to the `patches` folder. Just like [nixpkgs-python](https://github.com/cachix/nixpkgs-python/tree/main/patches) does

Reply:
@pedorich-n makes sense, most likely this patch will not change over time.